scripts/detect_text.py

```diff
@@ -24,95 +24,71 @@
     if any(gpu_devices):
         tf.config.experimental.set_memory_growth(gpu_devices[0], True)
 
+IMAGE_FILE_EXTENSIONS = [".jpeg", ".jpg", ".png", ".tif", ".tiff", ".bmp"]
+OTHER_EXTENSIONS = [".pdf"]
 
-_OUTPUT_CHOICE_JSON = "json"
-_OUTPUT_CHOICE_TEXT = "txt"
 
+def _process_file(model, file_path: Path, out_format: str) -> None:
 
-def _process_file(model, file_path: Path, out_format: str) -> str:
-    if str(file_path).lower().endswith(".pdf"):
+    if out_format not in ["txt", "json", "xml"]:
+        raise ValueError(f"Unsupported output format: {out_format}")
+
+    if os.path.splitext(file_path)[1] in IMAGE_FILE_EXTENSIONS:
+        doc = DocumentFile.from_images([file_path])
+    elif os.path.splitext(file_path)[1] in OTHER_EXTENSIONS:
         doc = DocumentFile.from_pdf(file_path)
     else:
-        doc = DocumentFile.from_images(file_path)
+        print(f"Skip unsupported file type: {file_path}")
 
     out = model(doc)
-    export = out.export()
-
-    if out_format == _OUTPUT_CHOICE_JSON:
-        out_txt = json.dumps(export, indent=2)
-    elif out_format == _OUTPUT_CHOICE_TEXT:
-        out_txt = ""
-        for page in export["pages"]:
-            for block in page["blocks"]:
-                for line in block["lines"]:
-                    for word in line["words"]:
-                        out_txt += word["value"] + " "
-                    out_txt += "\n"
-            out_txt += "\n\n"
+
+    if out_format == "json":
+        output = json.dumps(out.export(), indent=2)
+    elif out_format == "txt":
+        output = out.render()
+    elif out_format == "xml":
+        output = out.export_as_xml()
+
+    path = Path("output").joinpath(file_path.stem + "." + out_format)
+    if out_format == "xml":
+        for i, (xml_bytes, xml_tree) in enumerate(output):
```

Comment:
getting a warning here about an unused variable

Reply:
Hi 👋 yeah we could replace it with _ but so it is more clear what it returns :)

Reply:
The XML export returns a list which contains a tuple (XML as bytes, the whole ElementTree) for each page :)

Reply:
ok clear

thanks a lot for the cleanup work on the PR :-)

Reply:
No problem 🤗